src/MartinGeorgiev/Doctrine/DBAL/Types/Point.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MartinGeorgiev\Doctrine\DBAL\Types;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use MartinGeorgiev\Doctrine\DBAL\Types\Exceptions\InvalidPointForDatabaseException;
+use MartinGeorgiev\Doctrine\DBAL\Types\Exceptions\InvalidPointForPHPException;
+use MartinGeorgiev\Doctrine\DBAL\Types\ValueObject\Point as PointValueObject;
+
+/**
+ * Implementation of PostgreSQL POINT data type.
+ *
+ * @see https://www.postgresql.org/docs/17/datatype-geometric.html#DATATYPE-GEOMETRIC-POINTS
+ * @since 3.1
+ *
+ * @author Sébastien Jean <sebastien.jean76@gmail.com>
+ */
+class Point extends BaseType
+{
+    protected const TYPE_NAME = 'point';
+
+    public function convertToDatabaseValue($value, AbstractPlatform $platform): ?string
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        if (!$value instanceof PointValueObject) {
+            throw InvalidPointForPHPException::forInvalidType($value);
+        }
+
+        return (string) $value;
+    }
+
+    public function convertToPHPValue($value, AbstractPlatform $platform): ?PointValueObject
+    {
+        if ($value === null) {
+            return null;
+        }
+
+        if (!\is_string($value)) {
+            throw InvalidPointForDatabaseException::forInvalidType($value);
+        }
+
+        if (!\preg_match(PointValueObject::POINT_REGEX, $value, $matches)) {
+            throw InvalidPointForDatabaseException::forInvalidFormat($value);
+        }
+
+        return new PointValueObject((float) $matches[1], (float) $matches[2]);
```

Comment:
```suggestion

        try {
            return PointValueObject::fromString($value);
        } catch (\InvalidArgumentException) {
            throw InvalidPointForDatabaseException::forInvalidFormat($value);
        }
```

This will be a more elegant solution that encapsulates the validation logic in the VO only, as suggested [here](https://github.com/martin-georgiev/postgresql-for-doctrine/pull/348#discussion_r2054948713) by the AI. All the constants then can become private and there won't be need to expose them.

Reply:
Done :)